src/DataTransferObject/LoginHintDto.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\DataTransferObject;
+
+use InvalidArgumentException;
+
+class LoginHintDto
+{
+    /** @var string */
+    private $username;
+
+    /** @var string */
+    private $groupId;
+
+    /** @var string */
+    private $slug;
+
+    public function __construct(string $username, string $groupId, string $slug)
```

Comment:
I like the concept of having a VO for login hint but I'm afraid you've picked the wrong parameters here.

What do we want to know from a login hint?
- Who is the user (= `username` ✔️ )
- Which assignment are we trying to launch ( = assignment ➖ )

The line item is not specific enough information, because a user can have multiple assignments for the same line item. With your current implementation we would generate the same login hint for all of them.

Also group id is redundant information here, the username identifies the user already, so it can be removed.

Reply:
Changed to have username and assignment id